src/main/webapp/app/course/course-scores/course-scores-student-statistics.ts

```diff
@@ -10,10 +10,14 @@ export class CourseScoresStudentStatistics {
     numberOfParticipatedExercises = 0;
     numberOfSuccessfulExercises = 0;
     overallPoints = 0;
-    pointsPerExercise = new Map<number, number>(); // the index is the exercise id
-    sumPointsPerExerciseType = new Map<ExerciseType, number>(); // the absolute number (sum) of points the students received per exercise type
-    scorePerExerciseType = new Map<ExerciseType, number>(); // the relative number of points the students received per exercise type (divided by the max points per exercise type)
-    pointsPerExerciseType = new Map<ExerciseType, number[]>(); // a string containing the points for all exercises of a specific type
+    // the index is the exercise id
+    pointsPerExercise = new Map<number, number>();
+    // the absolute number (sum) of points the students received per exercise type
+    sumPointsPerExerciseType = new Map<ExerciseType, number>();
+    // the relative number of points the students received per exercise type (divided by the max points per exercise type)
+    scorePerExerciseType = new Map<ExerciseType, number>();
+    // the points for all exercises of a specific type; for each type the exercise titles are mapped to the number of points
```

Comment:
What do you think about using the id, as the name does not always have to be unique :thinking:

Reply:
I thought about that, too. In this component that is not an issue, however, as the exercises titles are modified on initialization to ensure uniqueness: https://github.com/ls1intum/Artemis/blob/bb2639fd20fad4d38858bafe57c26c93a005f8e8/src/main/webapp/app/course/course-scores/course-scores.component.ts#L185

Reply:
I agree, but using a title as an identifier when there is an id looks strange to me :thinking: But we can keep it like this as a quick fix :+1:

Reply:
I had another look and refactored the code to now use the ids instead of the titles as map keys.